torch/_subclasses/fake_tensor.py

```diff
@@ -284,14 +284,17 @@ def mk_fake_tensor(make_meta_t):
         return out
 
     # If you specify the device, it MUST be a meta tensor.
-    def from_meta_and_device(self, fake_mode, t, device):
+    def from_meta_and_device(self, fake_mode, t, device, keep_leafness=False):
         assert (
             t.device.type == "meta"
         ), f"tensor's device must be `meta`, got {t.device.type} instead"
         maybe_memo = self._get_memo(t)
         if maybe_memo is not None:
             return maybe_memo
-        out = FakeTensor(fake_mode, t, device)
+        if keep_leafness and not t.is_leaf:
+            out = FakeTensor(fake_mode, t, device).clone()
```

Comment:
@ezyang Is there a good way of preserving `is_leaf` from `t` when we construct a `FakeTensor` around it?

Reply:
You can look at how MetaConverter does it very carefully, but in general it's pretty difficult to do using the preexisting API surface in public PyTorch.